TeachingRecordSystem/src/TeachingRecordSystem.Core/Events/ProfessionalStatusUpdatedEvent.cs

```diff
@@ -0,0 +1,35 @@
+using TeachingRecordSystem.Core.Events.Models;
+using File = TeachingRecordSystem.Core.Events.Models.File;
+
+namespace TeachingRecordSystem.Core.Events;
+
+public record ProfessionalStatusUpdatedEvent : EventBase, IEventWithPersonId, IEventWithProfessionalStatus, IEventWithKey
+{
+    public string? Key { get; init; }
+    public required Guid PersonId { get; init; }
+    public required ProfessionalStatus ProfessionalStatus { get; init; }
+    public required ProfessionalStatus OldProfessionalStatus { get; init; }
+    public required string? ChangeReason { get; init; }
+    public required string? ChangeReasonDetail { get; init; }
+    public required File? EvidenceFile { get; init; }
+    public required ProfessionalStatusUpdatedEventChanges Changes { get; init; }
+}
+
+[Flags]
+public enum ProfessionalStatusUpdatedEventChanges
+{
+    None = 0,
+    ProfessionalStatus = 1 << 0, // CML TODO needed?
```

Comment:
I'm unclear about this ProfessionalStatus, which is derived from the route, but also stored in the model.